mrp_project/models/hr_analytic_timesheet.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# (c) 2016 Daniel Dico <dd@oerp.ca>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api
+
+
+class HrAnalyticTimesheet(models.Model):
+    _inherit = "hr.analytic.timesheet"
+
+    @api.model
+    def create(self, vals):
+        vals['mrp_production_id'] = 'mrp_production_id' in vals and \
+                                    vals['mrp_production_id'] or \
```

Comment:
if instead of using `vals['mrp_production_id']` you use `vals.get('mrp_production_id, False)` you won't require to check `'mrp_production_id' in vals`